### PR TITLE
generic.tests.jumbo: update test steps.

### PR DESCRIPTION
**Pull request by weliao**
generic.tests.jumbo: update test steps.

1.  update test steps.
2. delete windows guest test for e1000 Nic.

ID:1361026

Signed-off-by: Wei Liao weliao@redhat.com

### CODE REVIEW

**Comment by vivianQizhu** [comment]
13:06:19 INFO | Ping 10.66.9.236 with size 4578
13:06:19 INFO | The command of Ping is: ping 10.66.9.236  -c 1 -s 4578 -M do
13:06:19 DEBUG| PING 10.66.9.236 (10.66.9.236) 4578(4606) bytes of data.
13:06:19 DEBUG| 4586 bytes from 10.66.9.236: icmp_seq=1 ttl=64 time=0.154 ms

LGTM, Acked-by: Qianqian Zhu qizhu@redhat.com

File: generic/tests/jumbo.py [hunk 1]
```diff
@@ -118,13 +127,13 @@ def verify_mtu():
...
             logging.info("Flood with large frames")
-            utils_test.ping(guest_ip, interface=ifname,
+            utils_test.ping(guest_ip,
                             packetsize=max_icmp_pkt_size,
                             flood=True, timeout=float(flood_time))
 
         def large_frame_ping(count=100):
             logging.info("Large frame ping")
-            _, output = utils_test.ping(guest_ip, count, interface=ifname,
+            _, output = utils_test.ping(guest_ip, count,
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: same as line 110.

File: generic/tests/jumbo.py [hunk 2]
```diff
@@ -98,14 +107,14 @@ def run(test, params, env):
         utils.run(arp_add_cmd)
 
         def is_mtu_ok():
-            status, _ = utils_test.ping(guest_ip, 1, interface=ifname,
+            status, _ = utils_test.ping(guest_ip, 1,
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: how about status = utils_test.ping(guest_ip, 1,
                                                       packetsize=max_icmp_pkt_size,
                                                       hint="do", timeout=2)[0]

File: generic/tests/jumbo.py [hunk 3]
```diff
@@ -49,6 +50,14 @@ def run(test, params, env):
     if guest_ip is None:
         raise error.TestError("Could not get the guest ip address")
 
+    host_ifaces = utils_net.get_host_iface()
+    host_mtu_cmd = "ifconfig %s mtu %s"
+    logging.info("Change all host NICs MTU to 9000")
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: I think it's better to use "logging.info("Change all host NICs MTU to %s" % mtu)"

File: generic/tests/jumbo.py [hunk 4]
```diff
@@ -176,3 +184,9 @@ def size_increase_ping(step=random.randrange(90, 110)):
         if utils.system("grep '%s.*%s' /proc/net/arp" % (guest_ip, ifname)) == '0':
             utils.run("arp -d %s -i %s" % (guest_ip, ifname))
             logging.info("Removing the temporary ARP entry successfully")
+
+        logging.info("Change back host NICs MTU to 1500")
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: same as line 55

File: generic/tests/cfg/jumbo.cfg [hunk 5]
```diff
@@ -8,16 +8,16 @@
         mtu = 1500
     e1000:
         Linux:
-            mtu = 16110
+            mtu = 9000
```
Comment (PandaWei) [inline generic/tests/cfg/jumbo.cfg]: why 9000? can you explain it?

Comment (PandaWei) [inline generic/tests/cfg/jumbo.cfg]: And if you have strong reason to adopt 9000,
why not make it as a default value and declare it below "type = jumbo"?

Comment (weliao) [inline generic/tests/cfg/jumbo.cfg]: Now need set switch all port to the same MTU, as bridge , the Maximum value only can set 9000. you can see https://bugzilla.redhat.com/show_bug.cgi?id=1348105#c7

Comment (weliao) [inline generic/tests/cfg/jumbo.cfg]: some other NIC can't used 9000, like rtl8139, spapr-vlan

File: generic/tests/jumbo.py (hunk 1 above)
Comment (weliao) [inline generic/tests/jumbo.py]: this test Loss ratio of large frame ping

File: generic/tests/jumbo.py (hunk 4 above)
Comment (weliao) [inline generic/tests/jumbo.py]: ?

File: generic/tests/jumbo.py (hunk 3 above)
Comment (weliao) [inline generic/tests/jumbo.py]: I've update,thanks for your suggest

File: generic/tests/jumbo.py (hunk 2 above)
Comment (weliao) [inline generic/tests/jumbo.py]: this is old code, seems no much effect, thanks for your suggest.

File: generic/tests/jumbo.py (hunk 1 above)
Comment (weliao) [inline generic/tests/jumbo.py]: thanks for your suggest.

File: generic/tests/jumbo.py (hunk 4 above)
Comment (weliao) [inline generic/tests/jumbo.py]: thanks for your suggest.

File: generic/tests/cfg/jumbo.cfg (hunk 5 above)
Comment (weliao) [inline generic/tests/cfg/jumbo.cfg]: I've update,thanks for your suggest

**Comment by weliao** [comment]
@pingl2016 @PandaWei  updated it, could you help to review again, thank you.

File: generic/tests/jumbo.py [hunk 6]
```diff
@@ -49,6 +50,14 @@ def run(test, params, env):
     if guest_ip is None:
         raise error.TestError("Could not get the guest ip address")
 
+    host_ifaces = utils_net.get_host_iface()
+    host_mtu_cmd = "ifconfig %s mtu %s"
+    logging.info("Change all host NICs MTU to %s" % mtu)
+    for iface in host_ifaces:
+        if iface == "switch":
```
Comment (xutian) [inline generic/tests/jumbo.py]: Bridge and ovs interface name not always switch. This change will make other user in trouble.
I guest you want to:
`target_ifaces = set(utils_net.get_host_iface()) - set([params["netdst"]])`

**Comment by xutian** [comment]
@weliao And, wait for your comments and other guys patch.

File: generic/tests/jumbo.py (hunk 6 above)
Comment (weliao) [inline generic/tests/jumbo.py]: updated it, thanks for your suggest,could you help to review again, thank you.

**Comment by pingl2016** [comment]
(4/7) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo:  PASS (522.65 s)
 (5/7) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.jumbo:  PASS (564.45 s)

Acked-by: Ping Li pingl@redhat.com

**Comment by weliao** [comment]
@pingl2016 updated,could you help to review again, thank you.

File: generic/tests/jumbo.py [hunk 7]
```diff
@@ -49,6 +50,26 @@ def run(test, params, env):
...
+        ifaces_in_use = host_bridges.list_iface()
+        target_ifaces = set(ifaces_in_use) - set(br_in_use)
+        logging.info("Change all Bridge NICs MTU to %s" % mtu)
+        for iface in target_ifaces:
+            utils.run(host_mtu_cmd % (iface, mtu))
+    else:
+        target_ifaces = get_ovs_ports(netdst)
+        logging.info("Change all Bridge NICs MTU to %s" % mtu)
+        for iface in target_ifaces:
+            utils.run(host_mtu_cmd % (iface, mtu))
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: I think  it's better to write as:
```
if ...:
    target_ifaces = ...
else:
    target_ifaces = ...
logging.info(...)
....
```

File: generic/tests/jumbo.py [hunk 8]
```diff
@@ -31,6 +31,7 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
     timeout = int(params.get("login_timeout", 360))
+    mtu_default = 1500
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: Does this default value change?

Comment (weliao) [inline generic/tests/jumbo.py]: yes, after test finish, will change back

File: generic/tests/jumbo.py (hunk 7 above)
Comment (weliao) [inline generic/tests/jumbo.py]: updated it, thanks for your suggest,could you help to review again, thank you.

**Comment by pingl2016** [comment]
case failed with nic e1000 on windows guest  
02:20:59 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py:420
02:20:59 ERROR| Traceback (most recent call last):
02:20:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 204, in runTest
02:20:59 ERROR|     raise exceptions.TestError(details)
02:20:59 ERROR| TestError: MTU is not as expected even after 10 seconds    [context: Add a temporary static ARP entry ...]
02:20:59 ERROR|
02:20:59 ERROR| ERROR 2-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.e1000.Win7.x86_64.sp1.io-github-autotest-qemu.jumbo -> TestError: MTU is not as expected even after 10 seconds    [context: Add a temporary static ARP entry ...]

**Comment by CongLi** [comment]
@weliao 

I think the commit log is not good, as we can not know what is your manual case in upstream,
 maybe "update case steps and $(some steps' descriptions)" is better.

File: generic/tests/jumbo.py [hunk 9]
```diff
@@ -49,6 +50,23 @@ def run(test, params, env):
...
+
+    host_mtu_cmd = "ifconfig %s mtu %s"
+    netdst = params.get("netdst")
+    host_bridges = utils_net.Bridge()
+    br_in_use = host_bridges.list_br()
+    if netdst in br_in_use:
+        ifaces_in_use = host_bridges.list_iface()
+        target_ifaces = set(ifaces_in_use) - set(br_in_use)
+    else:
+        target_ifaces = get_ovs_ports(netdst)
```
Comment (weliao) [inline generic/tests/jumbo.py]: update, thank you

**Comment by weliao** [comment]
@pingl2016 hi, I think case failed with nic e1000 on windows guest with my patch does not relate, it's another bug.

**Comment by weliao** [comment]
@CongLi  updated,could you help to review again, thank you.

**Comment by weliao** [comment]
JOB LOG    : /root/avocado/job-results/job-2016-10-18T14.36-b5d05f8/job.log
TESTS      : 4
 (1/4) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo: PASS (480.10 s)
 (2/4) smp_4.4096m.repeat2.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo: PASS (443.67 s)
 (3/4) smp_4.4096m.repeat3.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo: PASS (442.24 s)
 (4/4) smp_4.4096m.repeat3.post_check.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.check_coredump: PASS (13.66 s)

File: generic/tests/jumbo.py [hunk 10]
```diff
@@ -49,6 +50,23 @@ def run(test, params, env):
     if guest_ip is None:
         raise error.TestError("Could not get the guest ip address")
 
+    def get_ovs_ports(ovs):
```
Comment (pingl2016) [inline generic/tests/jumbo.py]: Doc string?

Comment (weliao) [inline generic/tests/jumbo.py]: list

**Comment by weliao** [comment]
@CongLi @xutian 

Could you help review this patch again?

Thanks

**Comment by pingl2016** [comment]
Acked-by: Ping Li <pingl@redhat.com>

 (22/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.rtl8139.RHEL.6.8.x86_64.io-github-autotest-qemu.jumbo:  PASS (842.47 s)
 (23/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.rtl8139.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo:  PASS (534.83 s)
 (24/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.rtl8139.Win10.x86_64.io-github-autotest-qemu.jumbo:  PASS (583.81 s)
 (25/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.rtl8139.Win7.x86_64.sp1.io-github-autotest-qemu.jumbo:  PASS (550.46 s)
 (26/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.rtl8139.Win2016.x86_64.io-github-autotest-qemu.jumbo:  PASS (595.02 s)
 (27/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.e1000.RHEL.6.8.x86_64.io-github-autotest-qemu.jumbo:  PASS (574.43 s)
 (28/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.e1000.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo:  PASS (547.89 s)
 (29/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.6.8.x86_64.io-github-autotest-qemu.jumbo:  PASS (570.53 s)
 (30/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.jumbo:  PASS (547.46 s)
 (31/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.jumbo:  PASS (634.40 s)
 (32/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.jumbo:  PASS (529.64 s)
 (33/33) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.jumbo:  PASS (580.93 s)

**Pull request merged**